src/client/main.ts

```diff
@@ -56,23 +69,50 @@ export interface AppIntent {
  * ```
  */
 export interface IntentResolution {
-    source: string;
+    /**
+     * The human-readable name of the app that resolved this intent.
+     */
+    source: AppName;
+    /**
+     * Any data returned by the intent.
+     */
     data?: unknown;
+    /**
+     * For future use. Right now always the string '1.0.0'.
+     */
     version: string;
 }
 
+/**
+ * Listener type alias. One of ContextListener or IntentListener.
+ */
 export type Listener = ContextListener | IntentListener;
 
+/**
+ * Listener for context broadcasts. Generated by [[addContextListener]].
+ */
 export interface ContextListener {
+    /**
+     * The callback for when this listener receives a context broadcast.
+     */
     handler: (context: Context) => void;
     /**
      * Unsubscribe the listener object.
```

Comment:
This could maybe do with some elaboration about what exactly it means to unsubscribe